docs/python/simctrl/cvode.rst

```diff
@@ -1256,12 +1256,53 @@ CVode
         membrane current out of the segment), is available
         at 0 area nodes (locations 0 and 1 of every section), does not require
         that extracellular be inserted (and so is much faster), and works
-        during parallel simuations with variable step methods. (ie. does not
+        during parallel simulations with variable step methods. (ie. does not
         require IDA which is currently not available in parallel).
-        i_membrane\_ exists as a range variable only when this function has
-        been called with an argument of 1.
+        i_membrane\_ exists as a range variable only when ``use_fast_imem`` has
+        been called with an argument of 1. Conversely, i_membrane\_ is
+        not computed when ``use_fast_imem`` is not called or with an
+        argument of 0.
+
+        i_membrane\_ include capacity current and all transmembrane
+        ionic currents but not stimulus currents. POINT_PROCESS synaptic
+        currents are considered ionic currents and so are included
+        in i_membrane\_. From charge conservation
+        a fundamental property is that the sum of all i_membrane\_ is
+        identical to the sum of all ELECTRODE_CURRENT (Current cannot
+        flow axially out of a cell since the root and leaves of each
+        cell tree have sealed end boundary conditions.)
+
+        The following tests this conservation law, assuming that the only
+        ELECTRODE_CURRENTs are IClamp. Note the idiom that visits all segments
+        of a model but only once each segment to sum up i_membrane\_
+
+        .. code:: python
 
-         
+            from neuron import h
+            h.CVode().use_fast_imem(1)
+
+            def assert_whole_model_charge_conservation():
+                # sum over all membrane current
+                total_imem = 0.0
+                for sec in h.allsec():
+                    for seg in sec.allseg(): # also the 0 area nodes at 0 and 1
+                        if seg.x == sec.orientation() and sec.parentseg() != None:
```

Comment:
Per [PEP 8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations),

> Comparisons to singletons like None should always be done with `is` or `is not`, never the equality operators.

Some of this is a style thing, but it's also possible to write `__neq__` such that non-`None` things could return `True` in this comparison.

Reply:
Thanks! I transposed that in my mind and made the mistake of thinking it was the opposite :)  But I'm not alone...
```
hines@hines-T7500:~/neuron/fastimem$ grep -r '== None' * | wc
     32     220    2407
```
I'll change it the new places of this pull request.

Reply:
I opened #1259 for the existing places, including the related situation for "!= None".

Because the semantics are subtly different, running `black` as @Helveg plans to do will not change this. (I checked.)